src/components/Contacts/ContactDetails/ContactDetailsHeader/DeleteContactModal/DeleteContactModal.tsx

```diff
@@ -22,15 +25,71 @@ interface DeleteContactModalProps {
   setOpen: (open: boolean) => void;
   deleting: boolean;
   deleteContact: () => void;
+  contactId: string;
 }
 
 export const DeleteContactModal: React.FC<DeleteContactModalProps> = ({
   open,
   setOpen,
   deleting,
   deleteContact,
+  contactId,
 }) => {
   const { t } = useTranslation();
+  const accountListId = useAccountListId() ?? '';
+  const [canDelete, setCanDelete] = useState(true);
+
+  const { data } = useContactSourceQuery({
+    variables: { accountListId, contactId },
+    skip: !open && !contactId,
+  });
+  const contactSources = data?.contact;
+
+  useEffect(() => {
+    if (!contactSources) {
+      return;
+    }
+    // We ensure the contact was created on MPDX and that all the data is editable.
+    // If any data is not editable, this means it was created by a third party.
+    // Which will only recreate the data after deleting it on MPDX.
+    // To prevent this confusion, we do not allow a contact to be deleted if it has non editable data.
+
+    const isContactNonEditable = !isEditableSource(contactSources.source ?? '');
+
+    const isAddressNonEditable = contactSources.addresses?.nodes.some(
+      (address) => !isEditableSource(address.source ?? ''),
+    );
+
+    const hasNonEditablePersonData = contactSources.people?.nodes?.map(
```

Comment:
These checks will also prevent users from deleting contacts with information that was imported from Google (and possibly TNT). Do we want that?